gemrb/includes/SDLInitSubSystemWrapper.h

```diff
@@ -0,0 +1,69 @@
+/* GemRB - Infinity Engine Emulator
+ * Copyright (C) 2024 The GemRB Project
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ *
+ */
+
+#ifndef SDLINITSUBSYSTEMWRAPPER_H
+#define SDLINITSUBSYSTEMWRAPPER_H
+
+#include <cstdlib>
+#include <SDL.h>
+
+#include "Logging/Logging.h"
+
+namespace GemRB {
+
+static bool SDLInitSubSystemWrapper(Uint32 flags) {
+	const char* owner = nullptr;
+
+	// FIXME: doesn't handle multiple subsystems in flags
+
+	switch (flags) {
+	case SDL_INIT_AUDIO:
+		owner = "SDLAudio";
+		break;
+	case SDL_INIT_GAMECONTROLLER:
+		owner = "SDLGamecontroller";
+		break;
+	case SDL_INIT_JOYSTICK:
+		owner = "SDLJoystick";
+		break;
+	case SDL_INIT_VIDEO:
+		owner = "SDLVideo";
+		break;
+	default:
+		owner = "UNIMPLEMENTED";
+		break;
+	}
```

Comment:
This will fail so rarely, I don't see much value in pretty printing it (there are more readable errors down the line anyway). You can just use `"SDL Init: {}", flags` instead of the owner.

Reply:
Ok.